core/src/main/java/com/distkv/core/block/Block.java

```diff
@@ -0,0 +1,210 @@
+package com.distkv.core.block;
+
+import com.distkv.common.utils.ByteUtil;
+import sun.nio.ch.DirectBuffer;
+
+import java.nio.ByteBuffer;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Block it's mainly used to store the data. such as key, value, metadata and so on.
+ *
+ * for non-fixed length data. stored as blew format. the pointer is integer type.
+ * | pointer1 | pointer2 | ...... | data2 | data1 |
+ * the pointer be used to quick find the data start offset and length.
+ *
+ * for non-fixed length data with key and value. stored as blew format.
+ * | key1 pointer | value1 pointer |  ...... | value1 data | key1 data |
+ *
+ * for fixed length data
+ * | data1 | data2 | data3 | data4 | ...... |
+ *
+ */
+public class Block {
+
+  private final ByteBuffer buffer; //buffer use direct memory to store data.
+  private final int capacity;
+  private int size; // only used by non fixed value.
+  private long blockId; // may be needed.
+
+  public Block(int capacity) {
+    checkArgument(capacity > 0);
+    this.capacity = capacity;
+    this.buffer = ByteBuffer.allocateDirect(this.capacity);
+  }
+
+  public void clean() {
+    ((DirectBuffer) buffer).cleaner().clean();
+  }
+
+  public byte[] read(int offset, int length) {
+    buffer.position(offset);
+    byte[] valueBytes = new byte[length];
+    buffer.get(valueBytes);
```

Comment:
i will modify for this

Reply:
oh, only modify for method such as getInt(offset), getShort(offset) and so on.because the offset is not the index of buffer which bytes to read.

```
// The offset within the array of the first byte to be read;
buffer.get(valueBytes, offset, length);
```